vscode-eslint/compile-eslint-server.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+# @see https://github.com/microsoft/vscode-eslint
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+REPO_DIR="${SCRIPT_DIR}"
+SRC_DIR="${REPO_DIR}/checkout"
+SRC_SERVER_DIR="${SRC_DIR}/server"
+DIST_DIR="${REPO_DIR}/out"
+
+
+# ------------------------- #
+# download the source codes #
+# ------------------------- #
+
+pushd "${REPO_DIR}" || exit
+
+rm -rf \
+    "${SRC_DIR}" "${DIST_DIR}" \
+    "package.json" "package-lock.json"
+
+echo 'Enter commit SHA or tag (for example v2.1.0) to build'
+read -p 'SHA or tag: ' ref
+
+git clone https://github.com/microsoft/vscode-eslint "${SRC_DIR}"
+
+popd || exit
+
+# ------------ #
+# prepare deps #
+# ------------ #
+
+pushd "${SRC_SERVER_DIR}" || exit
+
+git checkout $ref || exit
+
+npm install
+npm install -D @types/node
+
+# ------- #
+# compile #
+# ------- #
+
+npx tsc --newLine LF -p .
```

Comment:
I seem to failed on this step with

```
error TS6047: Argument for '--target' option must be 'ES3', 'ES5', or 'ES6'.
error TS5023: Unknown compiler option 'moduleResolution'.
error TS5023: Unknown compiler option 'lib'.
```

Reply:
It seems to be cause I have a outdated global typescript (`tsc`) install. We should have a module-level one?

```bash
npm install -D typescript @types/node
```

Reply:
Added typescript

Reply:
I cannot mark a conversation as resolved btw.